### PR TITLE
Adding check for minimum nodejs version

### PR DESCRIPTION
**Pull request by joeldenning**
See related https://github.com/single-spa/create-single-spa/issues/100

### CODE REVIEW

File: packages/create-single-spa/bin/create-single-spa.js [hunk 1]
```diff
@@ -1,8 +1,36 @@
 #!/usr/bin/env node
 
+checkNodeVersion();
+
 const yeoman = require("yeoman-environment");
 const argv = require("yargs").argv;
 
 const env = yeoman.createEnv();
 env.registerStub(require("generator-single-spa"), "single-spa");
-env.run(`single-spa ${argv._.join(" ")}`, argv);
```
Comment (joeldenning) [inline packages/create-single-spa/bin/create-single-spa.js:8]: who knows - maybe someone will be using a version of node that doesn't support template strings. Poor souls

File: packages/create-single-spa/bin/create-single-spa.js [hunk 2]
```diff
@@ -1,8 +1,36 @@
...
+  } else if (currentMajor === minMajor) {
+    if (currentMinor < minMinor) {
+      validVersion = false;
+    } else if (currentMinor === minMinor && currentPatch < minPatch) {
+      validVersion = false;
+    }
+  }
+
+  if (!validVersion) {
+    throw Error("create-single-spa requires NodeJS >= " + minVersion);
```
Comment (filoxo) [inline packages/create-single-spa/bin/create-single-spa.js:34]: I find it very helpful when the tool also shows what it received vs what is expected. What do you think of adding the detected version of node to this error message?

Comment (TheMcMurder) [reply packages/create-single-spa/bin/create-single-spa.js:34]: Great idea!

Comment (joeldenning) [reply packages/create-single-spa/bin/create-single-spa.js:34]: updated

**Pull request merged**